File: _examples/ups/csharp/Ups.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.IO;
using Microsoft.CSharp;
using Newtonsoft.Json;

namespace UPSScrapey
{
	class MainClass
	{
		// Please note that you'll need Json.Net
		// http://james.newtonking.com/pages/json-net.aspx
		static void Main (string[] args)
		{
			// Scrapey URL
			const string SCRAPEY = "http://localhost:8100/c/ups-blog/";
			// Initial page to start scraping
			var q = new Queue<string> ();
			q.Enqueue ("http://blog.ups.com/");

			// Max number of results pages to scrape
			var maxResultsPages = 5;

			// Please note that this code scrapes one page at a time
			while (q.Count > 0) {
				// Make a request to scrapey
				var url = SCRAPEY + "?url=" + Uri.EscapeUriString(q.Dequeue ());
				var res = HttpWebRequest.Create (new Uri (url)).GetResponse();
				var bodyStr = new StreamReader (res.GetResponseStream ()).ReadToEnd ();
				// The result is JSON, so parse this
				dynamic body = JsonConvert.DeserializeObject(bodyStr);

				// Depending on what kind of page we have
				switch (res.Headers.Get ("Scrapey-Handler")) {
				case "ups-result":
					// We either add more pages to the queue (detail links and previous page)
					if (--maxResultsPages > 0)
						q.Enqueue (body["previousPage"].Value);
					foreach (var p in body.posts)
						q.Enqueue (p["link"].Value);
					break;
				case "ups-detail":
					// On the detail page we can now consume the data we got
					Console.WriteLine (body.title + " (" + body.author + ") has " + body ["comments"].Count + " comments");
					break;
				}
			}
		}
	}
}

File: _examples/ups/node/ups.js
// Run |npm install async request| first
var async = require('async');
var request = require('request');

// The URL to your api'alizer instance
var APIALIZER = 'http://localhost:8100/c/ups-blog/';
var maxResultPages = 5;

var q = async.queue(function(url, next) {
  // Make the request to api'alizer
  request.get(APIALIZER + '?url=' + encodeURIComponent(url), function(err, res, body) {
    // Check the err variable and res.statusCode here

    // We'll get a JSON response so parse the body
    body = JSON.parse(body.toString('utf8'));
    // On the results page push all the links to post to the queue (+ prev page)
    if (res.headers['scrapey-handler'] === 'ups-result') {
      body.posts.forEach(function(p) { q.push(p.link); });
      if (--maxResultPages > 0) {
        body.previousPage && q.push(body.previousPage);
      }
    }
    else if (res.headers['scrapey-handler'] === 'ups-detail') {
      // We now have detail page, let's do something with it...
      console.log(body.title + ' (' + body.author + ') has ' + body.comments.length + ' comments');
    }
    next();
  });
}, 4); // 4 requests simultaneous

// Initial URL that we'll start scraping
q.push('http://blog.ups.com/');
